doc/howto/uefi-secureboot-testing.rst

```diff
@@ -0,0 +1,1222 @@
+            from /etc/secureboot/keys/KEK/microsoft-kekca-public.der.siglist.KEK.signed
+          /CN=test-key
+            from /etc/secureboot/keys/KEK/test-cert.der.siglist.KEK.signed
+        db:
+          /C=US/ST=Washington/L=Redmond/O=Microsoft Corporation/CN=Microsoft Corporation UEFI CA 2011
+            from /etc/secureboot/keys/db/microsoft-uefica-public.der.siglist.db.signed
+          /C=US/ST=Washington/L=Redmond/O=Microsoft Corporation/CN=Microsoft Windows Production PCA 2011
+            from /etc/secureboot/keys/db/microsoft-pca-public.der.siglist.db.signed
+        dbx:
+      New keys in filesystem:
+        /etc/secureboot/keys/db/microsoft-pca-public.der.siglist.db.signed
+        /etc/secureboot/keys/db/microsoft-uefica-public.der.siglist.db.signed
+        /etc/secureboot/keys/KEK/test-cert.der.siglist.KEK.signed
+        /etc/secureboot/keys/KEK/microsoft-kekca-public.der.siglist.KEK.signed
+        /etc/secureboot/keys/PK/test-cert.der.siglist.PK.signed
+      Inserting key update /etc/secureboot/keys/db/microsoft-pca-public.der.siglist.db.signed into db
+      Inserting key update /etc/secureboot/keys/db/microsoft-uefica-public.der.siglist.db.signed into db
+      Inserting key update /etc/secureboot/keys/KEK/test-cert.der.siglist.KEK.signed into KEK
+      Inserting key update /etc/secureboot/keys/KEK/microsoft-kekca-public.der.siglist.KEK.signed into KEK
+      Inserting key update /etc/secureboot/keys/PK/test-cert.der.siglist.PK.signed into PK
+      Skipping bootloader signing for 'microsoft'
+
+#. Reboot and verify Secure Boot is enabled and configured like :ref:`bootloader-signed-with-user-key`, except with:
+
+   - 2 different keys in KEK (should match GUIDs for KEK from sb-setup)
+   - 2 different keys in db (should match GUIDs for db from sb-setup)
+
+#. :guilabel:`Boot from File` normally (notice :file:`shimx64.efi` and :file:`grubx64.efi` are listed -- this is the :file:`grubx64.efi` as installed by :command:`sudo grub-install --uefi-secure-boot`. I.e., the one signed with Canonical's key).
+
+#. Verify the machine booted with Secure Boot (you can also use :command:`sbsigdb` from newer versions of ``sbsigntool`` if it is available):
+
+   .. code:: none
+
+        sudo fwts uefidump - | grep Secure
+        Name: [[SecureBoot]].
+          Value: 0x01 (Secure Boot Mode On).
+
+#. Reboot and try to boot :file:`grubx64.efi` (i.e., the one signed with Canonical's key). This should fail to boot (when you press :kbd:`Enter` to select it, nothing happens).
+
+
+.. _bootloader-signed-with-canonical-key:
+
+Bootloader signed with Canonical key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This emulates a configuration that supports machines with Canonical's key in KEK and DB (user key still in PK and KEK). The key database is configured with (user key in PK and KEK has the same GUID, Canonical key in KEK and db has the same GUID):
```

Comment:
Canonical's key should be only used via shim, and not via KEK and DB currently, suggesting this is insecure.